### PR TITLE
Fix Typo: Rename header to headers  In Instrumentation Middleware 

### PR DESCRIPTION
**Pull request by ZhujiahuiFighting**
Previously, changes `header` to `headers`. But instrumentation middleware was forgotten. So it failed to get request id. So log's request id mismatches.

### CODE REVIEW

**Review by HangtianWu — APPROVED** [review]

File: packages/overreact/src/middleware/instrumentation/instrumentation-utils.js [hunk 1]
```diff
@@ -178,4 +178,4 @@ export function errorHandler(
     pass: instrumentationContext.requestResult,
     statusCode: error.status,
   });
-}
+}
```
Comment (zlike) [inline packages/overreact/src/middleware/instrumentation/instrumentation-utils.js:181]: Why no newline?

**Review by zlike — APPROVED** [review]
Approving but please add the newline back :)

File: packages/overreact/src/middleware/instrumentation/instrumentation-utils.js (hunk 1 above)
Comment (ZhujiahuiFighting) [reply packages/overreact/src/middleware/instrumentation/instrumentation-utils.js:181]: eslint? I will revert it

Comment (ZhujiahuiFighting) [reply packages/overreact/src/middleware/instrumentation/instrumentation-utils.js]: Updated.

**Pull request merged**